packages/editor/src/editor/sidebar/fields/browser/useAttributeBrowser.tsx

```diff
@@ -3,28 +3,47 @@ import { IvyIcons } from '@axonivy/ui-icons';
 import { useMeta } from '../../../../context/useMeta';
 import { findAttributesOfType, fullVariablePath, variableTreeData, findVariablesOfType } from '../../../../data/variable-tree-data';
 import { useCallback, useEffect, useState } from 'react';
-import type { Variable } from '@axonivy/form-editor-protocol';
+import type { DataTable, Variable } from '@axonivy/form-editor-protocol';
 import { useAppContext } from '../../../../context/AppContext';
 import { useData } from '../../../../data/data';
+import type { onlyAttributeSelection } from '../../../../types/config';
 
 export const ATTRIBUTE_BROWSER_ID = 'Attribute' as const;
 
-export const useAttributeBrowser = (onlyAttributes: boolean, onlyTypesOf?: string): Browser => {
+export const useAttributeBrowser = (onlyAttributesFor?: onlyAttributeSelection, onlyTypesOf?: string): Browser => {
   const [tree, setTree] = useState<Array<BrowserNode<Variable>>>([]);
   const { context } = useAppContext();
   const variableInfo = useMeta('meta/data/attributes', context, { types: {}, variables: [] }).data;
-  const { element } = useData();
+  const { element, data } = useData();
   const dynamicList = element?.config.dynamicItemsList as string;
 
   useEffect(() => {
-    if (onlyTypesOf && !onlyAttributes) {
+    const findParentTableComponent = (): DataTable | undefined => {
+      for (const component of data.components) {
+        if (component.type === 'DataTable') {
+          const hasMatchingColumn = (component.config as unknown as DataTable).components.some(
+            childComponent => childComponent.id === element?.id
+          );
+          if (hasMatchingColumn) {
+            return component.config as unknown as DataTable;
+          }
+        }
+      }
+      return undefined;
+    };
+
+    const parentTableComponent = findParentTableComponent();
```

Comment:
This code is now called for every browser usage, even if it has nothing to do with a table...
This should only be executed if a column filter is active!